src/components/Output.tsx

```diff
@@ -88,6 +88,125 @@ type StyleProps = {
 };
 
 export type OutputProps = ElementProps & StyleProps;
+export type FormatNumberProps = ElementProps & { decimal?: string; group?: string }
+
+export const formatNumber = (params: FormatNumberProps) => {
```

Comment:
consider moving `formatNumber` to `lib/numbers.ts`

Reply:
may be worth adding a docstring/specify the naming here as well. there's probably going to be several ways we want to format numbers so this may not apply ubiquitously to all number formatting.

there's also a lot going on in there so it may be helpful for people to understand what this util fn does without having to go through all the code

Reply:
I can move it, but if we wanna do some good refactor it would also be good to move the type definition but if I do so, we need also to change all around the app the import for `OutputType`, `ShowSign`, ecc. (It's better to move type definition also to avoid import cycles).

Do you prefer if I do it now on in a different PR?

Reply:
IMO it's fine if we move it in a different PR